rhcephcompose/compose.py

```diff
@@ -111,37 +116,49 @@ def validate_builds_lists(self):
                                                           filename)
                         raise RuntimeError(msg)
 
-    @property
-    def output_dir(self):
+    def _generate_id(self, respin):
+        """
+        Generate a compose ID for this respin.
+
+        :param int respin: a respin number for this compose
+        :returns: a compose ID
+        """
+        compose_type = COMPOSE_TYPE_MAP[self.compose_type]
+        name_tmpl = '{release_short}-{release_version}-{oslabel}-{arch}-{date}{compose_type}.{respin}'  # NOQA
+        return name_tmpl.format(release_short=self.release_short,
+                                release_version=self.release_version,
+                                oslabel='Ubuntu',
+                                date=self.date,
+                                arch=self.arch,
+                                compose_type=compose_type,
+                                respin=respin)
+
+    def _find_respin(self):
         """
+        Find the next available respin number in self.target.
+
         Use the same logic that pungi/compose.py uses in order come up with
         the name for the output directory. The name should be something
         like "Ceph-1.3-Ubuntu-20160922.t.0-x86_64" to match what
         Pungi creates.
         """
-        if getattr(self, '_output_directory', None):
-            return self._output_directory
-        compose_date = time.strftime('%Y%m%d')
-        compose_name = '{release_short}-{release_version}-{oslabel}-{arch}-{compose_date}{compose_type}.{compose_respin}'  # NOQA
-        compose_type = COMPOSE_TYPE_MAP[self.compose_type]
-        compose_respin = 0
+        respin = 0
         while 1:
```

Comment:
I understand this `while 1:` was here before, but it makes me nervous to have an infinite loop like this. It will keep going as long as `output_dir` exists with a compose_id right? so why not make that the condition?

Reply:
You're totally right. I refactored this into two methods and I'll post this as a separate PR.

Reply:
refactor in https://github.com/red-hat-storage/rhcephcompose/pull/86